src/extension.ts

```diff
@@ -673,7 +673,18 @@ class ExtensionManager implements vscode.Disposable {
         }
         const drv = await cmt.getCMakeDriverInstance();
         const opts = drv ? drv.expansionOptions : undefined;
-        const env = await effectiveKitEnvironment(kit, opts);
+        let env: Map<string, string> = new Map<string, string>();
+        if (kit) {
+          env = await effectiveKitEnvironment(kit, opts);
+        } else if (cmt.configurePreset) {
+          for (const key in cmt.configurePreset.environment) {
+            const value = cmt.configurePreset.environment[key];
+            if (value) {
```

Comment:
Can we do an util.isString(value) here instead? I think some falsy checks in presets.ts also need to be fixed at some point.